lab4/provided/processor.cc

```diff
@@ -20,19 +21,15 @@ Processor::~Processor()
 void
 Processor::sendRequest(Record &r)
 {
-    uint8_t *data = nullptr;
     if (r.write) {
-        data = new uint8_t[r.size];
-        for (int i=0; i<r.size; i++) {
-            data[i] = (uint8_t)(r.requestId & 0xFF);
-        }
+        r.data = new uint8_t[r.size];
+        memset(r.data, (uint8_t)r.requestId, r.size);
```

Comment:
The memset change is good here, but I'd rather not push the other changes. I'm planning on adding some testing code that will use data soon.

Reply:
I've made some updates here now. Could you look at merging with the current master and see if this still makes sense. If so, I'm happy to merge it.

Reply:
The updates you made didn't affect this; it looks good to me.